UPCEANExtensions

Eism opened the pull request.

Thaodan commented:
Do you have fixup commits to fix commits with changes you previously added?
If so I'd smash those commits with the actual changes.
It looks like you have changes spread in multiple commits because of that.

ftylitak commented:
@Thaodan I not sure i understood 100% what you are trying to say. 
Do you suggest to change the already existing commits to be more descriptive or feature specific? 
I ask this because I am not sure which would  be the benefit of such a change at this point.  

Do I understand correctly?

ftylitak commented on src/QZXing.cpp:
```diff
@@ -398,7 +430,9 @@ QString QZXing::decodeImage(const QImage &image, int maxWidth, int maxHeight, bo
-                emit tagFoundAdvanced(string, foundedFmt, charSet_);
+
+                QVariantMap metadataMap = metadataToMap(res->getMetadata());
+                emit tagFoundAdvanced(string, foundedFmt, charSet_, metadataMap);
```
To avoid braking backward compatibility it would be better to emit both overloads of tagFoundAdvanced: 

```cpp
QVariantMap metadataMap = metadataToMap(res->getMetadata());
emit tagFoundAdvanced(string, foundedFmt, charSet_, metadataMap);
emit tagFoundAdvanced(string, foundedFmt, charSet_);
```

Eism commented:
When I added UPC_EAN_EXTENCION decoding, we could randomly decode UPC/EAN or UPC_EAN_EXTENCION due to line-by-line decoding(there could be no additional barcode in the first lines).
When I added the available additional barcode formats (2 or 5) and in the absence said that they did not decode, because of this we began to decode either UPC/EAN or UPC_EAN_EXTENCION.
To fix this, I added another decoding without an additional barcode.

Tests showed good performance(android). But UPC / EAN began to slow down a bit due to the omission of the first two decoding (there we do not find UPC_EAN_EXTENCION).
